src/Common/Filesystem.cpp

```diff
@@ -0,0 +1,233 @@
+//
+// Copyright (C) 2011-2017 Ben Key
+// Distributed under the Boost Software License, Version 1.0.
+// (See accompanying file LICENSE_1_0.txt or copy at
+// http://www.boost.org/LICENSE_1_0.txt)
+//
+
+#include <string>
+#include "Filesystem.h"
+#include "StringUtils.h"
+
+#if (BOOST_OS_CYGWIN || BOOST_OS_WINDOWS)
+#include "Internals/Windows.hpp"
+#endif
+
+namespace {
+    constexpr const char* pathSep =
+#if (BOOST_OS_WINDOWS)
+        ";";
+#else
+        ":";
+#endif
+
+    constexpr const char* sep =
+#if (BOOST_OS_WINDOWS)
+        "\\";
+#else
+        "/";
+#endif
+
+    std::string GetEnv(const std::string& varName) {
```

Comment:
Any reason to put env-related functions into filesystem.cpp?

Reply:
actually...see the fallback below